test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -10854,6 +10854,226 @@ public async Task RestoreNetCore_WhenPackageNamespacesConfiguredAndNoMatchingSou
             Assert.Contains($"Installed {packageY} {version} from {pathContext.PackageSource}", result.AllOutput);
         }
 
+        [Fact]
+        public async Task RestoreCommand_NameSpaceFilter_PR_WithAllRestoreSources_Properies_Succeed()
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            // Set up solution, project, and packages
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            var workingPath = pathContext.WorkingDirectory;
+            var opensourceRepositoryPath = Path.Combine(workingPath, "PublicRepository");
+            Directory.CreateDirectory(opensourceRepositoryPath);
+
+            var privateRepositoryPath = Path.Combine(workingPath, "PrivateRepository");
+            Directory.CreateDirectory(privateRepositoryPath);
+
+            var net461 = NuGetFramework.Parse("net461");
+
+            var projectA = new SimpleTestProjectContext(
+                "a",
+                ProjectStyle.PackageReference,
+                pathContext.SolutionRoot);
+            projectA.Frameworks.Add(new SimpleTestProjectFrameworkContext(net461));
+
+            // Add both repositories as RestoreSources
+            projectA.Properties.Add("RestoreSources", $"{opensourceRepositoryPath};{privateRepositoryPath}");
+
+            var packageOpenSourceA = new SimpleTestPackageContext()
+            {
+                Id = "Contoso.Opensource.A",
+                Version = "1.0.0"
+            };
+
+            packageOpenSourceA.AddFile("lib/net461/openA.dll");
+
+            var packageOpenSourceContosoMvc = new SimpleTestPackageContext()
+            {
+                Id = "Contoso.MVC.ASP",  // Package Id conflict with internally created package
+                Version = "1.0.0"
+            };
+            packageOpenSourceContosoMvc.AddFile("lib/net461/openA.dll");
+
+            var packageContosoMvcReal = new SimpleTestPackageContext()
+            {
+                Id = "Contoso.MVC.ASP",
+                Version = "1.0.0"
+            };
+            packageContosoMvcReal.AddFile("lib/net461/realA.dll");
```

Comment:
I think you can simplify the declaration for all packages. 

I'm sorry if this is nitpicky, but things are 3-4 times the length which makes it quite challenging to review them.